website_sale_order_type/models/sale_order.py

```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Simone Rubino - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.multi
+    def _cart_update(
+            self, product_id=None, line_id=None, add_qty=0, set_qty=0,
+            attributes=None, **kwargs):
+        self.ensure_one()
+        res = super(SaleOrder, self)._cart_update(
+            product_id=product_id, line_id=line_id,
+            add_qty=add_qty, set_qty=set_qty, attributes=attributes, **kwargs)
+        self.onchange_partner_id()
```

Comment:
This makes more than only assigning the sale order type. I think you should restrict this or put specifically the code for sale order type handling.

Reply:
Thanks for your review, done in https://github.com/OCA/e-commerce/pull/253/commits/8ff2990edeca6e79ca073c626e165fc0dfcff593